### PR TITLE
tcp_proxy: increase stat on upstream request retry

### PR DESCRIPTION
**Pull request by ohadvano**
Additional Description: increasing the ``upstream_rq_retry`` on retry in the ``tcp_proxy``
Risk Level: low
Testing: integration tests
Docs Changes: none
Release Notes: none
Platform Specific Features: none

### CODE REVIEW

**Review by botengyao — COMMENTED** [review]
Thanks @ohadvano for this improvement!

**Review by ggreenway — COMMENTED** [review]
This stat is currently documented (in `docs/root/configuration/upstream/cluster_manager/cluster_stats.rst`) as `upstream_rq_retry, Counter, Total request retries`. There's also `upstream_rq_retry_limit_exceeded`, which is not used by tcp_proxy (`upstream_cx_connect_attempts_exceeded` is used instead). Will it be confusing to use a `rq` stat instead of `cx` in this context? Maybe we should try to clarify in the docs at least. It may be better to create a new stat here.

File: source/common/tcp_proxy/tcp_proxy.cc [hunk 1]
```diff
@@ -457,6 +457,8 @@ Network::FilterStatus Filter::establishUpstreamConnection() {
     cluster->trafficStats()->upstream_cx_connect_attempts_exceeded_.inc();
     onInitFailure(UpstreamFailureReason::ConnectFailed);
     return Network::FilterStatus::StopIteration;
+  } else if (connect_attempts_ >= 1) {
```
[the remark this answers is not in the document]
Comment (ohadvano) [reply source/common/tcp_proxy/tcp_proxy.cc:460]: If ``max_connect_attempts`` equals 1, there'll be no retry, so I think it's legit to not increment this stat?

**Comment by ohadvano** [comment]
I agree, I'll pivot to adding ``upstream_cx_retry`` instead to keep it consistent

**Comment by ramaraochavali** [comment]
but is n't this tracked by `upstream_cx_connect_fail` though/

**Comment by ohadvano** [comment]
@ramaraochavali does this stat increase for each upstream connection failure (i.e retries count) or is it per downstream connection?

**Comment by ggreenway** [comment]
Once for each upstream failure

**Pull request closed without merging**